File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [1]*(n+1)
        res[0]=0
        x=1
        while(x):
            for i in range((2**x)+1,(2**(x+1))):
                if i>n:
                    x=0
                    break
                res[i]=res[i-2**x]+1
            if(not x):
                break
            x+=1
        return res

File: Data Structures & Algorithms/island-perimeter/submission-1.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        
        int res = 0;

        int l = grid.length;
        int m = grid[0].length;

        for(int i=0; i<l; i++){
            for(int j=0; j<m; j++){
                res += 4 - check(i,j,grid);
            }
        }

        return res;
    }

    private int check(int i, int j, int[][] grid){
        if (grid[i][j] == 0){
            return 4;
        }
        int curr = 0;
        if(i>0 && grid[i-1][j]==1){
            curr += 2;
        }
        if(j>0 && grid[i][j-1]==1){
            curr += 2;
        }

        return curr;
        
    }
}